utils/uri/matcher.go

```diff
@@ -0,0 +1,75 @@
+package uri
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	ErrInvalidURI = errors.New("invalid URI string")
+)
+
+//hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+type URI struct {
+	path      string
+	arguments map[string]string
+	metadata  map[string]string
+}
+
+// hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+func ParseURI(source string) (*URI, error) {
```

Comment:
A comment for this public method could be useful as well.

Reply:
Yep, these are just placeholders at the moment, I'll add some more detailed comments